src/java.base/share/classes/java/io/FileInputStream.java

```diff
@@ -42,18 +42,12 @@
  *
  * @apiNote
  * To release resources used by this stream {@link #close} should be called
- * directly or by try-with-resources. Subclasses are responsible for the cleanup
- * of resources acquired by the subclass.
- * Subclasses that override {@link #finalize} in order to perform cleanup
- * should be modified to use alternative cleanup mechanisms such as
- * {@link java.lang.ref.Cleaner} and remove the overriding {@code finalize} method.
+ * directly or by try-with-resources.
```

Comment:
I wonder if we should re-word the API note at the same time. I think it would be more readable to start with "The close method should be called ...".

Reply:
Sounds good to me.